app/src/main/java/swati4star/createpdf/util/PermissionsUtils.java

```diff
@@ -0,0 +1,63 @@
+package swati4star.createpdf.util;
+
+import android.app.Activity;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PermissionsUtils {
+    private static final String TAG = "PermissionsUtils";
+
+    /**
+     * checkRuntimePermissions takes in an Activity instance, a permission request code and
+     * multiple permissions or an array of permission and checks for if the permission is
+     * granted and if not it requests for the desired permissions as a group. The try and
+     * catch block checks for invalid permissions passed.
+     *
+     * @param activity
+     * @param requestCode
+     * @param permissions
+     * @return
+     */
+    public static boolean checkRuntimePermissions(Activity activity, int requestCode, String... permissions) {
+        List<String> listPermissionsNeeded = new ArrayList<>();
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            for (int i = 0; i < permissions.length; i++) {
+                if ((ContextCompat.checkSelfPermission(activity.getApplicationContext(),
+                        permissions[i])
+                        != PackageManager.PERMISSION_GRANTED)) {
+                    listPermissionsNeeded.add(permissions[i]);
+                }
+            }
+            try {
+                requestRuntimePermissions(activity, listPermissionsNeeded, requestCode);
+                return false;
```

Comment:
Can you explain the reason behind this false?

Reply:
If the permssions are not granted it checks for the permission and returns false for the time. Because in some fragments it uses a boolean return value that is used in some functioning in a particular fragment.

Reply:
@jddeep @sidhuparas @srv-twry 
 it will return true only when` Build.VERSION.SDK_INT < Build.VERSION_CODES.M`. otherwise it will always return false whether permission is granted ot not. correct me if I am wrong.